lib/src/expression_builder.dart

```diff
@@ -59,18 +85,74 @@ abstract class ExpressionBuilder implements CodeBuilder<Expression> {
 
   const ExpressionBuilder._();
 
+  /// Return a new [ExpressionBuilder] invoking the result of this expression.
+  ExpressionBuilder invokeSelf(
+    String name, {
+    Iterable<CodeBuilder<Expression>> positional: const [],
+    Map<String, CodeBuilder<Expression>> named: const {},
+  }); // TODO(matanl): Rename to invoke when factory is removed.
+
   /// Returns wrapped as a [ExpressionFunctionBody] AST.
   ExpressionFunctionBody toFunctionBody() => _asFunctionBody(this);
 
   /// Returns wrapped as a [FunctionExpression] AST.
   FunctionExpression toFunctionExpression() => _asFunctionExpression(this);
+
+  /// Returns as a [StatementBuilder].
```

Comment:
nit: for doc comment, how about "Converts to" rather that "Returns as" to underscore that it's not a simple cast but a conversion.